### PR TITLE
fix: handle illegal state exceptions in TopicMessageQuery subscription

### LINKED ISSUE #1344: [Beginner]: Handle empty mirror network safely in TopicMessageQuery subscription
## 🐥 Beginner Friendly

This issue is a great fit for contributors who are ready to explore the Hiero C++ codebase a little more and take on slightly more independent work.

Beginner Issues often involve reading existing C++ code, understanding how different parts of the SDK fit together, and making small, thoughtful updates that follow established patterns.

The goal is to support skill growth while keeping the experience approachable, well-scoped, and enjoyable.

---

## 👾 Description of the Issue

`TopicMessageQuery` can crash when the mirror network has no available nodes.

In:

```
src/sdk/main/src/TopicMessageQuery.cc
```

`getConnectedMirrorNode()` calls `network->getNextMirrorNode()`, then immediately checks `node->channelFailedToConnect()` in a loop. But `MirrorNetwork::getNextMirrorNode()` can return `nullptr` when the network is empty.

This leads to a null dereference before a helpful error is surfaced.

Relevant files:
```
src/sdk/main/src/TopicMessageQuery.cc
src/sdk/main/src/impl/MirrorNetwork.cc
```

---

## 💡 Proposed Solution

Make `TopicMessageQuery` handle an empty mirror network explicitly before dereferencing the node pointer.

The change should:
- Keep existing subscription behavior unchanged when nodes are available
- Provide a clear failure path (exception or error handler) when no mirror node is available
- Include a unit/integration test (or closest existing test pattern) for the empty-network scenario

---

## 👩‍💻 Implementation Steps

- [ ] Open `src/sdk/main/src/TopicMessageQuery.cc`
- [ ] Review `getConnectedMirrorNode()` and the subscription startup flow
- [ ] Add a null check after `getNextMirrorNode()` and before dereferencing
- [ ] Ensure the no-node case exits with a clear, actionable error instead of crash
- [ ] Review `src/sdk/main/src/impl/MirrorNetwork.cc` to confirm behavior when no nodes are configured
- [ ] Add or extend tests to cover empty mirror network handling
- [ ] Build the SDK and ensure all tests pass

---

## ✅ Acceptance Criteria

- [ ] **Scope:** Changes are limited to empty-network handling in topic subscription flow and related tests
- [ ] **Behavior:** Existing behavior unchanged when mirror nodes are available
- [ ] **Tests:** Existing tests pass; empty-network handling is covered
- [ ] **Review:** All code review feedback addressed

---

## 📋 Step-by-Step Contribution Guide

- [ ] Comment `/assign` to request the issue
- [ ] Wait for assignment
- [ ] Fork the repository and create a branch
- [ ] Set up the project using the instructions in `README.md`
- [ ] Make the requested changes
- [ ] Sign each commit using `-s -S`
- [ ] Push your branch and open a pull request

Read [Workflow Guide](https://github.com/hiero-ledger/hiero-sdk-cpp/blob/main/docs/training/workflow.md) for step-by-step workflow guidance.
Read [README.md](https://github.com/hiero-ledger/hiero-sdk-cpp/blob/main/README.md) for setup instructions.

❗ Pull requests **cannot be merged** without `S` and `s` signed commits.
See the [Signing Guide](https://github.com/hiero-ledger/hiero-sdk-cpp/blob/main/docs/training/signing.md).

---

## �� Additional Information

If you have questions while working on this issue, feel free to ask!

You can reach the community and maintainers here:
[Hiero-SDK-C++ Discord](https://discord.com/channels/905194001349627914/1337424839761465364)

Whether you need help finding the right file, understanding existing code, or confirming your approach — we are happy to help.

### PR DESCRIPTION
**Pull request by oGranny**
**Description**:
improves error handling in the `TopicMessageQuery` subscription logic, ensuring that cases where the mirror network is not configured or no mirror nodes are available are handled gracefully. It also adds a new integration test to verify this behavior.

**Related issue(s)**:
Fixes #1344

**Checklist**

- [x] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: src/sdk/tests/integration/TopicMessageQueryIntegrationTests.cc [hunk 1]
```diff
@@ -269,3 +270,32 @@ TEST_F(TopicMessageQueryIntegrationTests, CanReceiveLargeTopicMessage)
   ASSERT_NO_THROW(const TransactionReceipt txReceipt =
                     TopicDeleteTransaction().setTopicId(topicId).execute(getTestClient()).getReceipt(getTestClient()));
 }
+
+//-----
+TEST_F(TopicMessageQueryIntegrationTests, SubscribeWithEmptyMirrorNetwork)
+{
+  // Given
+  Client client = Client::fromConfigFile("local_node.json");
+  client.setMirrorNetwork({});
```
Comment (rwalworth) [inline src/sdk/tests/integration/TopicMessageQueryIntegrationTests.cc:279]: **suggestion (non-blocking):** The test fixture's `SetUp()` already initializes a client from `local_node.json` - you could reuse it here instead of constructing a new one:

```cpp
Client client = getTestClient();
client.setMirrorNetwork({});
```

This keeps it consistent with the pattern used by the rest of the test class and avoids duplicating the file path logic. Happy to merge as-is though and address this in a follow-up.

**Review by rwalworth — DISMISSED** [review]
LGTM @oGranny - the fix is clean and the logic is correct. I left one small non-blocking suggestion on the test below. Running workflows and will merge once they pass!

File: src/sdk/tests/integration/TopicMessageQueryIntegrationTests.cc (hunk 1 above)
Comment (oGranny) [reply src/sdk/tests/integration/TopicMessageQueryIntegrationTests.cc:279]: > Happy to merge as-is though and address this in a follow-up. 

Alright, Sounds good.

**Comment by oGranny** [comment]
Hey @rwalworth, I’ve fixed the linting issues. Let me know if everything looks good.

**Review by rwalworth — APPROVED** [review]
Linting fixes look clean - LGTM! Running the workflows and will merge once they pass!
